lnrpc/routerrpc/router_server.go

```diff
@@ -609,3 +613,45 @@ func (s *Server) BuildRoute(ctx context.Context,
 
 	return routeResp, nil
 }
+
+// SubscribeHtlcEvents creates a uni-directional stream from the server to
+// the client which delivers a stream of htlc events.
+func (s *Server) SubscribeHtlcEvents(req *SubscribeHtlcEventsRequest,
+	stream Router_SubscribeHtlcEventsServer) error {
+
+	htlcClient, err := s.cfg.RouterBackend.SubscribeHtlcEvents()
+	if err != nil {
+		return err
+	}
+	defer htlcClient.Cancel()
+
+	s.wg.Add(1)
+	defer s.wg.Done()
```

Comment:
hmm on second thought i suspect this is unnecessary since this doesn't spawn any _new_ goroutines. [`GracefulStop` will wait until all pending rpc calls are processed](https://github.com/grpc/grpc-go/blob/master/server.go#L1555)

Reply:
Yeah, felt weird to have it. So there are different `Stop` levels here? Stopping the internals of the subserver (achieved with quit) and stopping of the rpc server which happen separately.